internal/util/git/git.go

```diff
@@ -81,8 +84,16 @@ func GetAuthMethod(ctx context.Context, repoCred *apiv1.RepoCredential, kubeClie
 	return auth, insecureSkipTLS, nil
 }
 
+// GetReferenceName dynamically determines the reference type and returns the appropriate plumbing.ReferenceName only for tags and branches
+func GetReferenceName(reference string) plumbing.ReferenceName {
+	if reference[0] == 'v' || reference[0] == 'V' {
+		return plumbing.NewTagReferenceName(reference)
```

Comment:
Do Tags always need to start with "v" or "V"? Should we maybe have the user specify whether it's a Tag or a Branch in the `GitSync` spec?

Reply:
If we decide that we should have this flexibility, I'm okay creating a follow up issue for that and just assuming for now.

Reply:
moreover, however, there can be branches that start with the letter "v"

Reply:
yeah i realised later this thing ,i am fixing this part .

Reply:
thanks

Reply:
Will it be better to have the targetRevision specified as: @juliev0 
- For tag - “tag/v1.0.0” 
- For branch - “branch_name”
?
Also do we need to support commit hash as well?

Reply:
Is there any built in support in go git to be able to tell this? If not what @chandankumar4 is suggesting is better.

Reply:
So, @chandankumar4 are you suggesting that we add a field to the `GitSync` spec or no? If we do include commit hash, there may not be a good way to discern that from branch name just by name alone, right?

Reply:
I’m suggesting to use the existing field “spec.targetRevision” and for commit hash it can be like: “commit/<hash>” ?

Reply:
Okay. I'll let @xdevxy weigh in on what she prefers.

Reply:
@shubhamdixit863 can you follow the logic like https://github.com/src-d/go-git/blob/master/repository.go#L1362-L1387, so we don't make any assumption of how the revision should look like? Thanks!

Reply:
@xdevxy i went through the link ,but the problem is for it to work i need a cloned repository  reference.And for cloning i need to have the proper referenceName .What i can do is do a shallow clone in memory and check for the reference then clone in the temp directory

Reply:
> @shubhamdixit863 can you follow the logic like https://github.com/src-d/go-git/blob/master/repository.go#L1362-L1387, so we don't make any assumption of how the revision should look like? Thanks!

This is valid only when have repository reference (pointer) before even cloning we need to figure out the branch or tag